src/csharp/Grpc.Core.Api/LiteClientBase.cs

```diff
@@ -0,0 +1,96 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using Grpc.Core.Utils;
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Base class for lightweight client-side stubs.
+    /// All calls are invoked via a <c>CallInvoker</c>.
+    /// Lite client stubs have no configuration knobs, all configuration
+    /// is provided by decorating the call invoker.
+    /// </summary>
+    public abstract class LiteClientBase
```

Comment:
So we are taking the new base class approach and have codegen generate this base class instead right? Will that be done through a command line arg?

Reply:
I don't know that yet, it's a difficult decision :-(, but it's definitely one of the options.
There are multiple things that come into play when talking about client side channels - context propagation, load balancing, credentials etc.
I keep hearing that we'll need to do a more detailed API review with other stakeholders from the gRPC team.

In parallel to that, I'm looking into resolving the problems with porting over CallOptions struct - currently I don't have a good answer there.

Reply:
Sounds good. Let us know when there's a more concrete design we should take a look at.

Reply:
The question still stands, Do we need to make the change to codegen?